crypto/src/hash.rs

```diff
@@ -291,25 +274,47 @@ impl<T: IntoSchema> IntoSchema for HashOf<T> {
     }
 }
 
+#[cfg(any(feature = "ffi_export", feature = "ffi_import"))]
+mod ffi {
+    //! Manual implementations of FFI related functionality
+
+    use super::*;
+
+    // NOTE: Hash is FFI serialized as an array (a pointer in a function call, by value when part of a struct)
+    iroha_ffi::ffi_type! {
+        unsafe impl Transparent for Hash {
+            type Target = [u8; Hash::LENGTH];
```

Comment:
do you think there could be any issue with treating Hash as transmutable into [u8; Hash::LENGTH] considering it's a repr(C) struct with two fields?

I'm slightly concerned about the effects on the calling convention, though probably that's not an issue

Reply:
It would be a problem under some circumstances, but nothing comes to mind. IMO, we shouldn't try to solve a problem before it appears.